### PR TITLE
Add testimonials design and API

### PR DESCRIPTION
**Pull request by iHiD**
See https://github.com/exercism/v3-project-management/issues/194

### CODE REVIEW

**Comment by iHiD** [comment]
@kntsoriano This is good, thanks.

I've added some commits with styling.

The PR needs rebasing and this needs fixing:
- The dropdown is open before the button is clicked again (see screenshot below). Can you ensure this doesn't happen in future PRs pls, because I'm going to miss it one time:

Screenshot 2021-04-12 at 13 15 11

File: app/commands/mentor/testimonial/retrieve.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        @criteria = criteria
+        @order = order || :newest
+        @track_slug = track_slug
+        @include_unrevealed = include_unrevealed
+      end
+
+      def call
+        setup!
+        filter!
+        filter_track!
```
Comment (ErikSchierboom) [inline app/commands/mentor/testimonial/retrieve.rb]: Maybe have `filter_track` be part of `filter`?

File: app/javascript/components/mentoring/testimonials-list/RevealedTestimonial.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    if (!isRevealed) {
+      return
+    }
+
+    setOpen(true)
+  }, [isRevealed])
+
+  return (
+    <React.Fragment>
+      <a
```
Comment (ErikSchierboom) [inline app/javascript/components/mentoring/testimonials-list/RevealedTestimonial.tsx:29]: Feel free to ignore this as I don't know much about accessibility, but is an anchor tag with a `#` href bad for accessibility? Maybe a button is better here? CC @SleeplessByte

File: app/javascript/components/mentoring/testimonials-list/TrackDropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    <React.Fragment>
+      <label className="c-radio-wrapper">
+        <input type="radio" onChange={onChange} checked={checked} />
+        <div className="row">
+          <div className="c-radio" />
+          <TrackIcon iconUrl={iconUrl} title={title} />
+          <div className="title">{title}</div>
+        </div>
+      </label>
+    </React.Fragment>
```
Comment (ErikSchierboom) [inline app/javascript/components/mentoring/testimonials-list/TrackDropdown.tsx]: I don't think you need the `React.Fragment`, as you already have one root element.

```suggestion
    <label className="c-radio-wrapper">
      <input type="radio" onChange={onChange} checked={checked} />
      <div className="row">
        <div className="c-radio" />
        <TrackIcon iconUrl={iconUrl} title={title} />
        <div className="title">{title}</div>
      </div>
    </label>
```

File: app/javascript/components/mentoring/testimonials-list/UnrevealedTestimonial.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+          }),
+        })
+
+        onRevealed()
+      },
+    }
+  )
+
+  return (
+    <React.Fragment>
```
Comment (ErikSchierboom) [inline app/javascript/components/mentoring/testimonials-list/UnrevealedTestimonial.tsx]: I don't think you need `React.Fragment` as there is one root element.

**Review by ErikSchierboom — APPROVED** [review]

File: app/javascript/components/mentoring/testimonials-list/RevealedTestimonial.tsx (hunk 2 above)
Comment (iHiD) [reply app/javascript/components/mentoring/testimonials-list/RevealedTestimonial.tsx:29]: Updated.

File: app/commands/mentor/testimonial/retrieve.rb (hunk 1 above)
Comment (iHiD) [reply app/commands/mentor/testimonial/retrieve.rb]: Updated.

File: app/javascript/components/mentoring/testimonials-list/UnrevealedTestimonial.tsx (hunk 4 above)
Comment (iHiD) [reply app/javascript/components/mentoring/testimonials-list/UnrevealedTestimonial.tsx]: Updated.

File: app/javascript/components/mentoring/testimonials-list/TrackDropdown.tsx (hunk 3 above)
Comment (iHiD) [reply app/javascript/components/mentoring/testimonials-list/TrackDropdown.tsx]: Updated :+1:

**Pull request merged**